### PR TITLE
TGUI Mining Processor

### PR DESCRIPTION
**Pull request by Geevies**
* Converted the ore processor UI to TGUI.

image

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: tgui/packages/tgui/interfaces/MiningProcessor.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+              />
+            </Box>
+          )}
+        </Section>
+        <Section title="Ore Processing">
+          <Table>
+            <Table.Row header>
+              <Table.Cell>Name</Table.Cell>
+              <Table.Cell>Stored</Table.Cell>
+              <Table.Cell>Processing</Table.Cell>
```
Comment (alsoandanswer) [inline tgui/packages/tgui/interfaces/MiningProcessor.tsx:61]:
```suggestion
              <Table.Cell>Setting</Table.Cell>
```

File: code/modules/mining/machine_processing.dm [hunk 2]
```diff
@@ -95,71 +97,55 @@
...
-
-	var/datum/browser/processor_win = new(user, "processor_console", capitalize_first_letters(name))
-	processor_win.set_content(dat)
-	processor_win.open()
-
-/obj/machinery/mineral/processing_unit_console/Topic(href, href_list)
+		var/processing_type = ""
+		switch(machine.ores_processing[ore])
+			if(0)
+				processing_type = "Not Processing"
```
Comment (alsoandanswer) [inline code/modules/mining/machine_processing.dm:130]:
```suggestion
				processing_type = "Idle"
```

File: code/modules/mining/machine_processing.dm [hunk 3]
```diff
@@ -95,71 +97,55 @@
...
+			if(1)
+				processing_type = "Smelting"
```
Comment (alsoandanswer) [inline code/modules/mining/machine_processing.dm:132]:
```suggestion
				processing_type = "Smelt"
```

File: code/modules/mining/machine_processing.dm [hunk 4]
```diff
@@ -95,71 +97,55 @@
...
+			if(2)
+				processing_type = "Compressing"
```
Comment (alsoandanswer) [inline code/modules/mining/machine_processing.dm:134]:
```suggestion
				processing_type = "Compress"
```

File: code/modules/mining/machine_processing.dm [hunk 5]
```diff
@@ -95,71 +97,55 @@
...
+			if(3)
+				processing_type = "Alloying"
```
Comment (alsoandanswer) [inline code/modules/mining/machine_processing.dm:136]:
```suggestion
				processing_type = "Alloy"
```

**Comment by alsoandanswer** [comment]
probably a good idea to force all the buttons to be the same width

**Comment by Geevies** [comment]
> probably a good idea to force all the buttons to be the same width

that'd be a bit of a pain to do, imo any cosmetic tweaks can be done after this series of TGUI PRs are merged

**Review by alsoandanswer — APPROVED** [review]

**Pull request merged**